internal/cloud/awscloud/awscloud.go

```diff
@@ -212,7 +212,7 @@ func WaitUntilImportSnapshotTaskCompletedWithContext(c *ec2.EC2, ctx aws.Context
 // The caller can optionally specify the boot mode of the AMI. If the boot
 // mode is not specified, then the instances launched from this AMI use the
 // default boot mode value of the instance type.
-func (a *AWS) Register(name, bucket, key string, shareWith []string, rpmArch string, bootMode *string) (*string, error) {
+func (a *AWS) Register(imageName string, bucket string, s3key string, tagName string, shareWith []string, rpmArch string, bootMode *string) (*string, error) {
```

Comment:
nitpick: there's no need to add type to each argument. This makes the line just longer...